src/rewrite.jl

```diff
@@ -268,31 +263,64 @@ function _is_decomposable_with_factors(ex)
 end
 
 """
-    rewrite(x)
+    rewrite(expr; assume_sums_are_linear::Bool = true) -> Tuple{Symbol,Expr}
+
+Rewrites the expression `expr` to use mutable arithmetics.
+
+Returns `(variable, code)` comprised of a `gensym`'d variable equivalent to
+`expr` and the code necessary to create the variable.
+
+## `assume_sums_are_linear`
```

Comment:
Bike-shedding is open for what we should call this.

Reply:
The way I see it is that it flattens the expression tree or that it tries to write it as a sum. What about `flatten_to_sum` ?

Reply:
Isn't a "product term" a "factor" ? What it's doing is distributing the factors so we could say `move_factors_into_sums` or `distribute_factors_into_sums`

Reply:
Updated to `move_factors_into_sums`